### DISCUSSION TITLE
504 on Vercel Deployment

### DISCUSSION THREAD

**Discussion by Gblur** [neuefische/questions#572] [Web] [answered]
Ive got a 504 Error when trying to fetch data from the mongo atlas DB.
This only happens in the deployed version.

Files:

This is where SWR fetches data
https://github.com/Gblur/capstone-project/blob/0170f384b87118ed8bb0252dd53dfb74b52df804/components/Dashboard/index.js

This is the api for that request
https://github.com/Gblur/capstone-project/blob/0170f384b87118ed8bb0252dd53dfb74b52df804/pages/api/maps/index.js

Error Message: 
Error 504
response: An error occurred with your deployment
FUNCTION_INVOCATION_TIMEOUT 

Deployment:
https://capstone-project-c4vlawni4-gblur.vercel.app/maps

Logs: 
![Screenshot 2023-04-13 at 21 42 44](https://user-images.githubusercontent.com/25586795/231866467-f1861f9c-371f-467e-a0c0-9a6ee4685e19.png)

Atlas Connection is stable.
As I mentioned, it works locally.

**Comment by toshydev** [comment]
If it works locally I wonder if you allowed access to the DB from everywhere like here:
https://github.com/neuefische/cgn-web-23-1/blob/main/sessions/backend-mongodb-atlas/assets/images/atlas_allow-access-from-anywhere.png

Otherwise it might be that your request is taking longer than allowed. Just found this SO thread:
https://stackoverflow.com/questions/67456980/getting-a-504-502-error-on-api-requests-in-nextjs-deployed-on-vercel

**Reply by Gblur** [reply]
sure, thats true.
Access is set to every ip address.
Environment Variables are also set in project settings.
Unfortunately I cannot raise the timeout time without a pro account, but that shouldnt be necessary because it cant be taken that long to get data from that cluster

**Comment by Gblur** [comment, marked as answer]
OK,
good news. Solved.
For anyone who wants to use MUI Icon ibrary. Be sure not to use named Imports from the whole Library.
Instead use something like `import HomeIcon from "@mui/material/<IconName>"`
